### PR TITLE
Test Filter Citizen

### PR DESCRIPTION
**Pull request by FaiellaFrancesco**
This pull request introduces widget tests for validating the functionality of the ‘comune’ and ‘search’ fields. Key highlights include:

- 	Validation logic to ensure the length of input does not exceed 255 characters for both fields.
- 	Test cases covering both valid and invalid scenarios for input length.
- 	Implementation of reusable test functions _testComune and _testSearch for efficient validation.

### CODE REVIEW

File: src/test/filtro_test.dart [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    // Verifica il risultato
+    expect(errorText, expected, reason: reason);
+
+    //print('$description:expected: $errorText reason:$reason');
+  });
+}
+
+
+/// Test for all the field of the form
```
Comment (Type-Here) [inline src/test/filtro_test.dart:154]: Dart Lint could warn about dead code here

**Review by Type-Here — APPROVED** [review]
Good Testing Documentation.
Input length check differs from solution adopted in other places but still a valid one.
Test on GUI elements should use in code widgets but a fidelity recreation of production code is still a valid option.
Lint could warn about dead code on multiple commented lines. A cleanup is suggested but not essential at this moment if pr passed CI/CD test.
Good Work :+1:

**Review by Luigi11162 — APPROVED** [review]
LGTM! Approved

**Review by benedettoscala — APPROVED** [review]

**Review by Borax-Gargamella — APPROVED** [review]
👍

**Pull request merged**